### PR TITLE
Finish updating github workflows and restructure them similar to othe…

### PR DESCRIPTION
**Pull request by svogt0511**
…r project workflows (lupo)

## Purpose

Finish updating and improve the github actions.  

## Approach

Uses a structure similar to lupo github actions.  Finished updating the github marketplace workflows to make sure they are using at least node 16.  Removed calls to the slack github marketplace action.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: .github/workflows/branch_to_staging.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  test:
+    uses: ./.github/workflows/ci.yml
+    secrets: inherit
+  call_build_and_push:
+    needs: test
+    uses: ./.github/workflows/build.yml
+    with:
+      image_name: ${{ github.repository }}
+      image_tag: main
```
Comment (jrhoads) [inline .github/workflows/branch_to_staging.yml:13]: Seems like the tag needs to be based on the branch, not main

File: .github/workflows/branch_to_staging.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    secrets: inherit
+  deploy:
+    needs: [test, call_build_and_push]
+    uses: ./.github/workflows/_update_terraform.yml
+    with:
+      image_tag: main
```
Comment (jrhoads) [inline .github/workflows/branch_to_staging.yml:19]: Same

File: .github/workflows/branch_to_staging.yml (hunk 1 above)
Comment (svogt0511) [reply .github/workflows/branch_to_staging.yml:13]: Great catch.  Done.

File: .github/workflows/branch_to_staging.yml (hunk 2 above)
Comment (svogt0511) [reply .github/workflows/branch_to_staging.yml:19]: Great catch.  Done.

**Review by jrhoads — APPROVED** [review]

**Pull request merged**